stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
+        'total_elevation_gain': 'meters',
+        'average_speed': 'm/s',
+        'max_speed': 'm/s'
+    }
 
-    elapsed_time = TimeIntervalAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.timedelta` of elapsed time for lap
-    moving_time = TimeIntervalAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.timedelta` of moving time for lap
-    start_date = TimestampAttribute(
-        (SUMMARY, DETAILED)
-    )  #: :class:`datetime.datetime` when lap was started in GMT
-    start_date_local = TimestampAttribute(
-        (SUMMARY, DETAILED), tzinfo=None
-    )  #: :class:`datetime.datetime` when lap was started local
-    distance = Attribute(
-        float, (SUMMARY, DETAILED), units=uh.meters
-    )  #: The distance for this lap.
-    start_index = Attribute(int, (SUMMARY, DETAILED))  #:
-    end_index = Attribute(int, (SUMMARY, DETAILED))  #:
-    total_elevation_gain = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters,
-    )  #: What is total elevation gain for lap
-    average_speed = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters_per_second,
-    )  #: Average speed for lap
-    max_speed = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters_per_second,
-    )  #: Max speed for lap
-    average_cadence = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average cadence for lap
-    average_watts = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average watts for lap
-    average_heartrate = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average heartrate for lap
-    max_heartrate = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Max heartrate for lap
-    lap_index = Attribute(int, (SUMMARY, DETAILED))  #: Index of lap
-    device_watts = Attribute(
-        bool, (SUMMARY, DETAILED)
-    )  # true if the watts are from a power meter, false if estimated
-    pace_zone = Attribute(int, (SUMMARY, DETAILED))  #: (undocumented)
-    split = Attribute(int, (SUMMARY, DETAILED))  #: Split number
+    _naive_local = validator('start_date_local', allow_reuse=True)(naive_datetime)
 
+    _activity_extension = extend_types('activity', model_class='Activity')
+    _athlete_extension = extend_types('athlete', model_class=Athlete)
 
-class Map(IdentifiableEntity):
-    id = Attribute(str, (SUMMARY, DETAILED))  #: Alpha-numeric identifier
-    polyline = Attribute(str, (SUMMARY, DETAILED))  #: Google polyline encoding
-    summary_polyline = Attribute(
-        str, (SUMMARY, DETAILED)
-    )  #: Google polyline encoding for summary shape
+
+class Map(PolylineMap):
```

Comment:
Going to keep looking here but is this the streaming data for activities? if so, why is map now an empty class? should i try to run / download the geo data from an activity?

Reply:
This is not the stream data (I haven't processed that entity yet). As mentioned above, this class is not empty: it inherits from `PolylineMap`. By subclassing it, we can use the name `Map` for this type to preserve backward compatibility.

Reply:
ok got it. thank you again!